tinman/sample.py

```diff
@@ -32,7 +32,7 @@ def main(argv):
         outfile = sys.stdout
     else:
         outfile = open(args.outfile, "w")
-    json.dump(snapshot, f, separators=(",", ":"))
+    json.dump(snapshot, outfile, separators=(",", ":"))
```

Comment:
`outfile` might be undefined here.

Reply:
I *think* `outfile` has definite-assignment.

Reply:
OIC. The horror of that code confused my brain...

```
if args.outfile == "-":
        outfile = sys.stdout
    else:
        outfile = open(args.outfile, "w")
```
Any idea why not
```
outfile = sys.stdout
if args.outfile != "-":
  outfile = open(args.outfile, "w")        
```

Reply:
Because there's no need to worry about scope.

Reply:
Entering into a block when there's no advantage seems wasteful. Extra code, extra logic.